scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -175,6 +176,196 @@ def _highly_variable_genes_seurat_v3(
         return df
 
 
+def _highly_variable_pearson_residuals(
+    adata: AnnData,
+    layer: Optional[str] = None,
+    n_top_genes: int = 2000,
+    batch_key: Optional[str] = None,
+    theta: float = 100,
+    clip: Union[Literal['auto', 'none'], float] = 'auto',
```

Comment:
`none` as literal is not optimal. I think this could be more `Optional[float] = None` and if `None` enters in "auto" mode?

Reply:
very nice, this makes everything much simpler. thx